File: src/main/java/no/nav/syfo/config/ApplicationConfig.java
package no.nav.syfo.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import org.springframework.jdbc.datasource.DataSourceTransactionManager;
import org.springframework.transaction.annotation.EnableTransactionManagement;

import javax.sql.DataSource;


@Configuration
@EnableTransactionManagement
@EnableJpaRepositories("no.nav.syfo")
public class ApplicationConfig {

    @Bean(name = "transactionManager")
    public DataSourceTransactionManager datasourceTransactionManager(DataSource dataSource) {
        DataSourceTransactionManager dataSourceTransactionManager = new DataSourceTransactionManager();
        dataSourceTransactionManager.setDataSource(dataSource);
        return dataSourceTransactionManager;
    }
}

File: src/main/kotlin/no/nav/syfo/config/ApplicationConfig.kt
package no.nav.syfo.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.transaction.annotation.EnableTransactionManagement
import org.springframework.data.jpa.repository.config.EnableJpaRepositories
import org.springframework.jdbc.datasource.DataSourceTransactionManager
import javax.persistence.EntityManagerFactory
import org.springframework.orm.jpa.JpaTransactionManager
import javax.sql.DataSource

@Configuration
@EnableScheduling
@EnableTransactionManagement
@EnableJpaRepositories("no.nav.syfo.repository")
class ApplicationConfig {
    @Bean
    fun datasourceTransactionManager(dataSource: DataSource?): DataSourceTransactionManager {
        val dataSourceTransactionManager = DataSourceTransactionManager()
        dataSourceTransactionManager.dataSource = dataSource
        return dataSourceTransactionManager
    }

    @Bean(name = ["transactionManager"])
    fun getTransactionManager(emf: EntityManagerFactory?, dataSource: DataSource?): JpaTransactionManager {
        val tm = JpaTransactionManager()
        tm.entityManagerFactory = emf
        tm.dataSource = dataSource
        return tm
    }
}
